src/codegen.rs

```diff
@@ -2818,6 +2818,12 @@ fn codegen_expr(indent: usize, expr: &CheckedExpression, project: &Project) -> S
                     output.push_str(&codegen_expr(indent, rhs, project));
                     output.push_str("; })");
                 }
+                BinaryOperator::NoneCoalescingAssign => {
+                    output.push_str(&codegen_expr(indent, lhs, project));
+                    output.push_str(".emplace(");
+                    output.push_str(&codegen_expr(indent, rhs, project));
```

Comment:
This looks like we will evaluate the RHS even if the LHS is `None`.
We'll want to short-circuit in that case, so that we don't get any side effects when assignment doesn't happen.

Reply:
I think you meant that if LHS has a value then we should short circuit, so I did that.